sklearn/utils/validation.py

```diff
@@ -499,14 +499,38 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning)
+
+    ## Indices Datatype regulation
+    sparse_indices_check(array,accept_large_sparse)
+
     return array
 
+def sparse_indices_check(array,accept_large_sparse=False):
+    """
+    Indices Regulation for CSR Matrices.
+    Only Int32 are supported for now
+
+    """
+    if accept_large_sparse==False:
+        if type(array) ==sp.csr.csr_matrix:
+            supported_indices=["int32"]
+            if hasattr(array, "indices"):
+                if hasattr(array.indices, "dtype"):
+                    if array.indices.dtype not in supported_indices:
+                        raise ValueError("CSR arrays accepts only 32 bit integers indexing. You provided %s bit indexing"
```

Comment:
I think just "Sparse matrices with 64bit indices are not supported"

Reply:
because even 8bit or other indices are also crashing, so that's why I added supported_indices array

Reply:
scipy.sparse should refuse to create <32bit integer indices.

Reply:
@jnothman I was able to create with  int8 and it gave segmentation fault in truncatedsvd, so I think we should add this matrix only